### PR TITLE
Remove unused functionality in cudf._lib.utils.pyx

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Contributes to https://github.com/rapidsai/cudf/issues/17317

More can be removed once my other cudf._lib PRs are in

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Review by vyasr — COMMENTED** [review]
@cwharris are any of the functions being removed here ones that you currently rely on in Morpheus?

File: python/cudf/cudf/_lib/utils.pyx
Comment (bdice) [inline python/cudf/cudf/_lib/utils.pyx:1]: If everything left is for pylibcudf interop, there may be a more descriptive name than “utils.” Feel free to punt on this.

**Review by bdice — APPROVED** [review]

File: python/cudf/cudf/_lib/utils.pyx
Comment (mroeschke) [reply python/cudf/cudf/_lib/utils.pyx:1]: Thanks. I think, if anything, the remaining used functions will be moved to `cudf/utils/ioutils.py` so I'll punt on the intermediate rename

**Comment by dagardner-nv** [comment]
In Morpheus we use the following which are being removed that we will need an alternative for:
* `data_from_unique_ptr`
We use this to construct a `cudf.DataFrame` from a `cudf::io::table_with_metadata`. Our current useage looks like:
```
data, index = data_from_unique_ptr(move(table.tbl), column_names=column_names, index_names=index_names)
df = cudf.DataFrame._from_data(data, index)
```

* `table_view_from_table` 
We use this when our C++ code receives a Python `cudf.DataFrame` object to get a libcudf `cudf::table_view` object which we can then use to read table/column data with.

**Comment by mroeschke** [comment]
Will not merge until https://github.com/nv-morpheus/Morpheus/pull/2082 is in

**Pull request merged**